ui/helpers/utils/state-corruption-html.ts

```diff
@@ -0,0 +1,134 @@
+import getFirstPreferredLangCode from '../../../shared/lib/get-first-preferred-lang-code';
+import { METHOD_REPAIR_DATABASE } from '../../../shared/lib/state-corruption-utils';
+import { t, updateCurrentLocale } from '../../../shared/lib/translate';
+
+export async function getStateCorruptionErrorHtml(
+  vaultRecoveryLink: string,
+  hasBackup: boolean,
+  currentLocale?: string,
+  supportLink?: string,
+) {
+  let preferredLocale: string;
+  if (currentLocale) {
+    preferredLocale = currentLocale;
+    await updateCurrentLocale(currentLocale);
+  } else {
+    preferredLocale = await getFirstPreferredLangCode();
+    await updateCurrentLocale(preferredLocale);
+  }
+
+  switchDirectionForPreferredLocale(preferredLocale);
+
+  const instructionsLink = `<a href="${lodashEscape(
+    vaultRecoveryLink,
+  )}" title="${lodashEscape(
+    t('stateCorruptionTheseInstructionsLinkTitle') ?? '',
+  )}" class="critical-error__link" target="_blank" rel="noopener noreferrer">${lodashEscape(
+    t('stateCorruptionTheseInstructions') ?? '',
+  )}</a>`;
+
+  let corruptionDetectedMessage: string,
+    copyAndRestoreMessage: string,
+    restoreOrResetMessage: string;
+  if (hasBackup) {
+    corruptionDetectedMessage = t('stateCorruptionDetectedWithBackup') ?? '';
+    copyAndRestoreMessage =
+      t('stateCorruptionCopyAndRestoreBeforeRecovery', instructionsLink) ?? '';
+    restoreOrResetMessage = t('stateCorruptionRestoreAccountsFromBackup') ?? '';
+  } else {
+    corruptionDetectedMessage = t('stateCorruptionDetectedNoBackup') ?? '';
+    copyAndRestoreMessage =
+      t('stateCorruptionCopyAndRestoreBeforeReset', instructionsLink) ?? '';
+    restoreOrResetMessage = t('stateCorruptionResetMetaMaskState') ?? '';
+  }
+
+  const header = `
+    <h1>${t('stateCorruptionMetamaskDatabaseCannotBeAccessed')}</h1>
+  `;
+  const body = `
+    <p>${lodashEscape(corruptionDetectedMessage)}</p>
+    <p>${copyAndRestoreMessage}</p>
+    <button disabled id="critical-error-button" class="critical-error__button-restore button btn-primary">
+      ${restoreOrResetMessage}
+    </button>
+  `;
+
+  const footer = supportLink
+    ? `<p class="critical-error__footer small">${lodashEscape(
+        t('unexpectedBehavior') ?? '',
+      )} <a
+        href="${lodashEscape(supportLink)}"
+        class="critical-error__link"
+        target="_blank"
+        rel="noopener noreferrer">
+          ${lodashEscape(t('sendBugReport') ?? '')}
+      </a>
+    </p>`
+    : '';
+
+  return getErrorHtmlBase(`
+    ${header}
+    ${body}
+    ${footer}
+  `);
+}
+
+export async function displayStateCorruptionError(
+  container: HTMLElement,
+  port: browser.Runtime.Port,
+  err: ErrorLike,
+  hasBackup: boolean,
+  currentLocale?: string,
+) {
+  log.error(err.stack);
+
+  function handleRestoreClick(this: HTMLButtonElement) {
+    // eslint-disable-next-line no-alert
+    const theyAreSure = confirm(t('stateCorruptionAreYouSure') ?? '');
```

Comment:
use of `confirm`. yay? nay?

Since we're not using React (this code piggy backs off existing critical error code that also uses raw HTML) for this stuff doing a more MM-on-brand prompt is a bit tricky.

Reply:
Yeah. `confirm` doesn't seem ideal, but it's a pragmatic choice here I think. I couldn't think of a compelling reason not to use it.